src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,107 @@
+package duke.parser;
+
+import duke.DukeException;
+import duke.command.*;
+import duke.common.Messages;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+
+public class Parser {
+    private static final String DEFAULT_DATE_FORMAT = "d/MM/yyyy HHmm";
+
+    public static Command parse(String userInput) throws DukeException {
+        String[] userInputArr = userInput.split(" ", 2);
+        switch (userInputArr[0]) {
+            case "mark":
+                return new MarkTaskCommand(Integer.parseInt(userInputArr[1]) - 1, true);
+            case "unmark":
+                return new MarkTaskCommand(Integer.parseInt(userInputArr[1]) - 1, false);
+            case "delete":
+                return new DeleteCommand(Integer.parseInt(userInputArr[1]) - 1);
+            case "find":
+                userInput = String.join(" ",
+                        Arrays.copyOfRange(userInputArr, 1, userInputArr.length));
+                return new FindTaskCommand(userInput);
+            case "list":
+                return new ListCommand();
+            case "bye":
+                return new ExitCommand();
+            case "todo":
+                return prepareAddTodoCommand(userInputArr[1]);
+            case "event":
+                userInput = String.join(" ",
+                        Arrays.copyOfRange(userInputArr, 1, userInputArr.length));
+                return prepareAddEventCommand(userInput);
+            case "deadline":
+                userInput = String.join(" ",
+                        Arrays.copyOfRange(userInputArr, 1, userInputArr.length));
+                return prepareAddDeadlineCommand(userInput);
+            default:
+                throw new DukeException(Messages.MESSAGE_ERROR_INVALID_COMMAND);
+        }
```

Comment:
Shouldn't the case indent be on the same level as the switch statement?
I noticed that this issue occurred a few times in your program. Do consider taking note of this.
```suggestion
        switch (userInputArr[0]) {
        case "mark":
            return new MarkTaskCommand(Integer.parseInt(userInputArr[1]) - 1, true);
        case "unmark":
            return new MarkTaskCommand(Integer.parseInt(userInputArr[1]) - 1, false);
        case "delete":
            return new DeleteCommand(Integer.parseInt(userInputArr[1]) - 1);
        case "find":
            userInput = String.join(" ",
                    Arrays.copyOfRange(userInputArr, 1, userInputArr.length));
            return new FindTaskCommand(userInput);
        case "list":
            return new ListCommand();
        case "bye":
            return new ExitCommand();
        case "todo":
            return prepareAddTodoCommand(userInputArr[1]);
        case "event":
            userInput = String.join(" ",
                    Arrays.copyOfRange(userInputArr, 1, userInputArr.length));
            return prepareAddEventCommand(userInput);
        case "deadline":
            userInput = String.join(" ",
                    Arrays.copyOfRange(userInputArr, 1, userInputArr.length));
            return prepareAddDeadlineCommand(userInput);
        default:
            throw new DukeException(Messages.MESSAGE_ERROR_INVALID_COMMAND);
        }
```